src/telemetry.rs

```diff
@@ -35,10 +43,68 @@ pub fn setup_logging() {
 #[cfg(not(feature = "console"))]
 pub fn setup_logging() {
     Lazy::force(&APPLICATION_START_TIME);
-    let filter_layer = tracing_subscriber::EnvFilter::try_from_default_env()
+    let filter = tracing_subscriber::EnvFilter::try_from_default_env()
         .or_else(|_| tracing_subscriber::EnvFilter::try_new("info"))
         .unwrap();
-    tracing_subscriber::registry()
-        .with(tracing_subscriber::fmt::layer().with_filter(filter_layer))
-        .init();
+    let (filter_layer, reload_handle) =
+        reload::Layer::new(tracing_subscriber::fmt::layer().with_filter(filter));
+    unsafe {
+        PRI_LOG_HANDLE = Some(Box::leak(Box::new(LogHandle(reload_handle))));
+    }
+    tracing_subscriber::registry().with(filter_layer).init();
+}
+
+// a handle to get and set the log level
+pub struct LogHandle(reload::Handle<FilteredLayer, Registry>);
+type BoxLayer = tracing_subscriber::fmt::Layer<tracing_subscriber::Registry>;
+pub(crate) type FilteredLayer = filter::Filtered<BoxLayer, EnvFilter, Registry>;
+
+pub fn set_level(level_str: String) -> Result<String, Error> {
```

Comment:
This onluy allows setting top level `level`; the library allows granular levels like `RUST_LOG=hbone::trace`. Should we allow more granular config?

Reply:
> This onluy allows setting top level `level`; the library allows granular levels like `RUST_LOG=hbone::trace`. Should we allow more granular config?

yes, will go for it

Reply:
hey John, can we split this work to another PR and merge this one firstly? @howardjohn